x-pack/plugins/cloud_security_posture/public/pages/findings/latest_findings/latest_findings_container.tsx

```diff
@@ -89,6 +92,49 @@ export const LatestFindingsContainer = ({ dataView }: FindingsBaseProps) => {
     });
   };
 
+  const flyoutFindingIndex = urlQuery?.findingIndex;
+
+  const pagination = getPaginationTableParams({
+    pageSize,
+    pageIndex: urlQuery.pageIndex,
+    totalItemCount: limitedTotalItemCount,
+  });
+
+  const onOpenFlyout = useCallback(
+    (flyoutFinding: CspFinding) => {
+      setUrlQuery({
+        findingIndex: slicedPage.findIndex(
+          (finding) =>
+            finding.resource.id === flyoutFinding?.resource.id &&
+            finding.rule.id === flyoutFinding?.rule.id
+        ),
+      });
+    },
+    [slicedPage, setUrlQuery]
+  );
+
+  const onCloseFlyout = () =>
+    setUrlQuery({
+      findingIndex: -1,
+    });
+
+  const onPaginateFlyout = useCallback(
+    (nextPageIndex: number) => {
+      const newPageIndex = Math.floor(nextPageIndex / pagination.pageSize);
+
+      const newFindingIndex =
+        nextPageIndex >= pagination.pageSize * newPageIndex
+          ? nextPageIndex - pagination.pageSize * newPageIndex
+          : nextPageIndex;
```

Comment:
maybe add a comment or use named constants so it would be easier to understand whats exactly is going on here.

Reply:
Thanks, I updated it to be simpler now!